core/templates/App.ts

```diff
@@ -205,11 +205,16 @@ angular.module('oppia').config([
               return $q(function(resolve, reject) {
                 // Get CSRF token before sending the request.
                 CsrfTokenService.getTokenAsync().then(function(token) {
-                  config.data = $.param({
-                    csrf_token: token,
-                    payload: JSON.stringify(config.data),
-                    source: document.URL
-                  }, true);
+                  if ((config.data instanceof FormData)) {
+                    config.data.append('csrf_token', token);
+                    config.data.append('source', document.URL);
```

Comment:
Why these changes? Can you please check how we have handled this for skill images? (This looks like a global change, can we do changes more specific to questions?)

Reply:
I've done this similar to how images in skills are handled. The only difference is that for skills this part is done using Angular 8  in the service `RequestInterceptor`

Reply:
Can we add a comment to explain the same?

\cc @bansalnitish @srijanreddy98 you might want to take a quick pass here.

Reply:
done